paymentapi/src/main/java/fi/hel/verkkokauppa/payment/repository/PaymentFilterRepository.java

```diff
@@ -0,0 +1,10 @@
+package fi.hel.verkkokauppa.payment.repository;
+
+import fi.hel.verkkokauppa.payment.model.PaymentFilter;
+import org.springframework.data.elasticsearch.repository.ElasticsearchRepository;
+
+import java.util.List;
+
+public interface PaymentFilterRepository extends ElasticsearchRepository<PaymentFilter, String> {
+    List<PaymentFilter> findAllByReferenceId(String referenceId);
```

Comment:
This search should use ReferenceType also.

Reply:
You are right, I'll update this.